pkg/storage/ingest/pusher.go

```diff
@@ -471,24 +492,28 @@ func newBatchingQueue(capacity int, batchSize int) *batchingQueue {
 
 // AddToBatch adds a time series to the current batch. If the batch size is reached, the batch is pushed to the Channel().
 // If an error occurs while pushing the batch, it returns the error and ensures the batch is pushed.
-func (q *batchingQueue) AddToBatch(ctx context.Context, ts mimirpb.PreallocTimeseries) error {
+func (q *batchingQueue) AddToBatch(ctx context.Context, source mimirpb.WriteRequest_SourceEnum, ts mimirpb.PreallocTimeseries) error {
```

Comment:
so these methods keep growing. I'm wondering if we shouldn't extract `(context.Context, mimirpb.WriteRequest_SourceEnum)` into the constructor of `batchingQueue` and use them from there? It also conveys the assumption that we can only use one of them at a time, and not every one which is passed to AddToBatch and AddMetadataToBatch actually makes a difference

Reply:
I think this is fine for now, 3 is still manageable - if we add the 4 one, maybe we should be looking at doing what you've said but for now, it's fine.